viadot/tasks/hubspot.py

```diff
@@ -0,0 +1,160 @@
+import json
+import pandas as pd
+import datetime
+
+from typing import List, Dict, Any, Literal
+from prefect import Task
+from prefect.tasks.secrets import PrefectSecret
+from prefect.utilities import logging
+from viadot.exceptions import ValidationError
+from viadot.sources import Hubspot
+
+logger = logging.get_logger()
+
+
+class HubspotToDF(Task):
+    def __init__(
+        self,
+        hubspot_credentials: dict,
+        *args,
+        **kwargs,
+    ):
+
+        self.credentials = hubspot_credentials
+        super().__init__(
+            name="hubspot_to_df",
+            *args,
+            **kwargs,
+        )
+
+    def __call__(self):
+        """Download Hubspot data to a DF"""
+        super().__call__(self)
+
+    def to_df(self, result: list = None) -> pd.DataFrame:
+        """
+        Args:
+            result (list, optional): API response in JSON format. Defaults to None.
+
+        Returns:
+            pd.DataFrame: DF from JSON
+        """
+        return pd.json_normalize(result)
+
+    def date_to_unixtimestamp(self, date: str = None):
+        """
+        Function for date conversion from user defined "yyyy-mm-dd" to Unix Timestamp (SECONDS SINCE JAN 01 1970. (UTC)).
+        For example: 1680774921 SECONDS SINCE JAN 01 1970. (UTC) -> 11:55:49 AM 2023-04-06.
+        """
+        clean_date = int(
+            datetime.datetime.timestamp(datetime.datetime.strptime(date, "%Y-%m-%d"))
+            * 1000
+        )
+        return clean_date
+
+    def format_filters(self, filters: Dict[str, Any] = {}) -> Dict[str, Any]:
+        """
+        Function for API body (filters) conversion from a user defined to API language. Converts date to Unix Timestamp.
+
+        Args:
+            filters (Dict[str, Any], optional): Filters in JSON format. Defaults to {}.
+
+        Returns:
+            Dict[str, Any]: Filters in JSON format after data cleaning
+        """
+        for item in filters:
+            for iterator in range(len(item["filters"])):
+                for subitem in item["filters"][iterator]:
+                    lookup = item["filters"][iterator][subitem]
+                    try:
+                        datetime.date.fromisoformat(lookup)
+                        date_to_format = item["filters"][iterator][subitem]
+                        date_after_format = self.date_to_unixtimestamp(date_to_format)
+                        filters[filters.index(item)]["filters"][iterator][
+                            subitem
+                        ] = date_after_format
+                    except:
```

Comment:
Why are you using a `try exception` block if you're not catching anything?

Reply:
Code is going through the list of strings defined in filters - if any of it will be a string date then it converts it, otherwise, skips it and goes deeper.